build.yaml

```diff
@@ -2424,6 +2422,40 @@ steps:
       - deploy_address_sa
       - address_image
       - create_certs
+  - kind: runImage
+    name: upload_query_jar
```

Comment:
Is there not a canonical place where this jar already sits? Not entirely clear to me yet why we need an extra bucket.

Reply:
`deploy_hail` uploads the JAR to `hail-common`, but that's only done on default deploy builds, not on dev or PR builds.

I'm generally inclined to not entangle the services team's infrastructure with compiler team's infrastructure.

The bucket is probably not necessary, but there wasn't another obvious place to put Hail Query Service stuff. Putting it in batch's bucket seems wrong.

Reply:
Sounds good, I'm fine with a separate bucket if it helps prevents awkward dependencies.